### PR TITLE
Feat: ThresholdCheckerMixin and WeightsDownloaderMixin

### PR DESCRIPTION
**Pull request by liyier90**
Changes:
- Make `Node` handle PeekingDuck specific requirements
  - Move RGB conversion functions to `Node`
  - Move hard coded "person" `bbox_labels` to `Node`
- Move `weights_utils` functions to `WeightsDownloaderMixin` and refactor
  - Update tests and use `mock.patch` decorator to improve test code readability
  - Add `no cover` for actual download and extract function (previously untested)
  - Use a `do_nothing` function instead of a fixture
- Rename files and class name following convention
  - `node.py` to `abstract_node.py`
  - Use snake case for file names
  - Use all caps for acronyms
- Implement `ThresholdCheckerMixin`
  - Expose a single `check_bounds` public method, with `method` and `include` arguments in a similar style to `pandas.fillna`'s `method` argument [ref](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.fillna.html)
    - Accepts a single `key` string or a list of `key` strings
    - Handles config values which are single values or a list of values
  - Expose a `check_valid_choice` public method
- Remove unused files and functions in YOLOv4 model (closes https://github.com/aimakerspace/PeekingDuck-Private/issues/70)

EDIT:
~~In progress (up for discussion)~~:
- Change internal `Detector`/`Predictor`/`Tracker` to take in individual config as arguments instead of config dict
  - Refactor YOLOX's `Detector`
  - Refactor all models' detector/predictor/tracker
  - Rearranged model configs to the following order:
    - inputs/outputs
    - weights
    - fixed configs (not meant for users to adjust)
    - user-adjustable configs
- Add groundtruths for object detection, pose estimation, and crowding counting models
  - Reformatted movenet's groundtruth to the standard format
  - Add function in conftest to load all groundtruths
- Refactor fixtures
  - Remove unnecessary prefixes
  - Shift movenet's single person and multi person fixtures here
    - Make HRNet and PoseNet also test on these fixtures
- Re-added `node.py` with deprecation warning through logger.

### CODE REVIEW

File: docs/source/tutorials/03_custom_nodes.rst [hunk 1]
```diff
@@ -174,7 +174,7 @@ implement our custom node function.
    
          from typing import Any, Dict, List, Tuple
          import cv2
-         from peekingduck.pipeline.nodes.node import AbstractNode
+         from peekingduck.pipeline.nodes.abstract_node import AbstractNode
```
[the remark this answers is not in the document]
Comment (liyier90) [reply docs/source/tutorials/03_custom_nodes.rst:177]: Re-added `node.py`. Using `logger.warning` to show deprecation warning now. 

Making `warnings` work require some more discussion such as:
- Where to show it: in my attempts it only shows at the start of the logs before all the other node initializations, causing it to get pushed up very quickly
- How to make it obvious: warning logs are colored yellow, but warnings.warn isn't. Should we try and color it and make it obvious? Will it make the output too cluttered/messy?
- Should we only show peekingduck warnings: some third party packages such as tensorflow also have deprecation warnings (like using `imp`). Choosing to show only warnings from PeekingDuck is not as straightforward when this package can be also used as a CLI tool
-

[the remark this answers is not in the document]
Comment (liyier90) [reply docs/source/tutorials/03_custom_nodes.rst:177]: Added a `deprecate()` function to display deprecation warning through `FutureWarning`.

Currently applied at `node.py`, `input.live/recorded` to `input.visual` conversion, and using `run_config.yml` as the default config path with `peekingduck run`.

File: tests/pipeline/nodes/test_base.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+        self.config = {}
+        self.logger = mock.Mock()
+
+
+class ThresholdModel(ThresholdCheckerMixin):
+    def __init__(self):
+        self.config = {"a": 12.5, "b": 10, "c": "value", "d": [10, 12.5, 13]}
+
+
+class TestBase:
```
Comment (leeping-ng) [inline tests/pipeline/nodes/test_base.py:49]: Minor point for consideration - perhaps can split into 2 classes, one for WeightsDownloaderMixin and one for ThresholdCheckerMixin.

File: peekingduck/pipeline/nodes/augment/brightness.py [hunk 3]
```diff
@@ -45,8 +46,7 @@ class Node(AbstractNode):
     def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
 
-        if not -100 <= self.beta <= 100:
-            raise ValueError("beta for brightness must be between [-100, 100]")
+        self.check_bounds("beta", (-100, 100), "within")
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/nodes/augment/brightness.py:49]: Default argument values is `"both"` so we are doing inclusive upper and lower bounds by default so most of the existing config thresholds are inclusive.

Only need to specific an argument value when we want to exclude any bounds.

But I can remove the default argument value if you feel that this need more clarity

File: peekingduck/pipeline/nodes/model/hrnet.py [hunk 4]
```diff
@@ -57,9 +57,6 @@ class Node(AbstractNode):
             Resolution of input array to HRNet model.
         score_threshold (:obj:`float`): **[0, 1], default = 0.1**. |br|
             Threshold to determine if detection should be returned
-        model_nodes (:obj:`Dict`):
-            **default = { inputs: [x:0], outputs: [Identity:0] }** |br|
-            Names of input and output nodes from model graph for prediction.
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/nodes/model/hrnet.py:62]: This is modifiable. Used in processing to change the aspect ratio and crop/resize the bbox regions before feeding into the model

File: tests/pipeline/nodes/test_base.py (hunk 2 above)
Comment (liyier90) [reply tests/pipeline/nodes/test_base.py]: Split to 2 test files

File: peekingduck/pipeline/nodes/augment/contrast.py [hunk 5]
```diff
@@ -43,8 +44,7 @@ class Node(AbstractNode):
     def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
         super().__init__(config, node_path=__name__, **kwargs)
 
-        if not 0 <= self.alpha <= 3.0:
-            raise ValueError("alpha for contrast must be between [0, 3]")
+        self.check_bounds("alpha", (0, 3), "within")
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/nodes/augment/contrast.py:47]: Based on discussions, using the default argument value is sufficient

File: peekingduck/pipeline/nodes/model/efficientdet_d04/efficientdet_model.py [hunk 6]
```diff
@@ -16,48 +16,49 @@
...
-        # check for efficientdet weights, if none then download into weights folder
-        weights_dir, model_dir = finder.find_paths(
-            config["root"], config["weights"], config["weights_parent_dir"]
-        )
-        if not checker.has_weights(weights_dir, model_dir):
-            self.logger.info("---no weights detected. proceeding to download...---")
-            downloader.download_weights(weights_dir, config["weights"]["blob_file"])
-            self.logger.info(f"---weights downloaded to {weights_dir}.---")
+        self.check_valid_choice("model_type", {0, 1, 2, 3, 4})
+        self.check_bounds("score_threshold", (0, 1), "within")
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/nodes/model/efficientdet_d04/efficientdet_model.py:42]: Based on discussions, using the default argument value is sufficient.

**Pull request merged**